### ISSUE TITLE
DataLoader converts cuda FloatTensor into cpu DoubleTensor when shape is (n,)

### ISSUE THREAD

**Issue by jpopham91** [pytorch/pytorch#2539]
In version 0.2.0, it appears that the DataLoader will convert a cuda.FloatTensor into a DoubleTensor when the shape is (n,)

```python
X = torch.rand(64,8).cuda()
y = torch.rand(64).cuda()
dataset = TensorDataset(X,y)
loader = DataLoader(dataset)
next((y for (X,y) in loader))
```
```
0.1309
[torch.DoubleTensor of size 1x1 (GPU 0)]
```

Using `unsqueeze` to convert the (n,) tensor to (n,1) fixes the issue.
```python
X = torch.rand(64,8).cuda()
y = torch.rand(64).cuda().unsqueeze(1)
dataset = TensorDataset(X,y)
loader = DataLoader(dataset)
next((y for (X,y) in loader))
```
```
0.8397
[torch.cuda.FloatTensor of size 1x1 (GPU 0)]
```
The workaround is easy enough, but I'm assuming this is a bug with either TensorDataset or DataLoader.

**Comment by skrish13** [comment]
this `TensorDataset` is from tnt/torchnet?

**Comment by jpopham91** [comment]
torch.utils.data.Dataset

**Comment by siarez** [comment]
I have similar issue where DataLoader returns a LongTensor where the original tensor was a cuda.LongTensor

**Comment by GaiYu0** [comment]
I encountered this problem as well.

**Comment by zzz123xyz** [comment]
I still have this problem which version r you using and solve this problem?

**Comment by ailzhang** [comment]
@zzz123xyz @GaiYu0 would you mind providing a minimal repro script for this on pytorch 1.1.0 or pytorch master? We will be willing to fix it if it still persists. Thanks!

**Comment by devforfu** [comment]
Is it still on the list? I have a similar problem when use in-memory NumPy dataset of float32 type but the data loader converts it into float64 after collation.

**Comment by FarzanT** [comment]
@devforfu Yes this problem still persists, and we have to (re)-convert back to float after getting a batch from DataLoader. I don't think that this is efficient.

**Comment by GabrielBianconi** [comment]
I am having the same issue :(
